cpp/oneapi/dal/algo/dbscan/backend/gpu/compute_kernel_brute_force_dpc.cpp

```diff
@@ -41,139 +44,270 @@ static result_t compute_kernel_dense_impl(const context_gpu& ctx,
+
+    const std::int64_t local_row_count = local_data.get_row_count();
+    const std::int64_t column_count = local_data.get_column_count();
 
-    const std::int64_t block_size = keeper.get_block_size();
-    const std::int64_t block_start = keeper.get_block_start();
-    const std::int64_t block_end = block_start + block_size;
-    const std::int64_t row_count = keeper.get_row_count();
-    auto arr_data = keeper.get_data();
-    auto arr_weights = keeper.get_weights();
+    std::int64_t global_row_count = local_row_count;
+    std::int64_t max_local_block_size = local_row_count;
+    {
+        ONEDAL_PROFILER_TASK(allreduce_rows_count_global, queue);
+        comm.allreduce(max_local_block_size, spmd::reduce_op::max).wait();
+    }
 
-    const double epsilon = desc.get_epsilon() * desc.get_epsilon();
+    auto send_recv_replace_local_size = array<std::int64_t>::zeros(1);
+    send_recv_replace_local_size.get_mutable_data()[0] = local_row_count;
+
+    auto global_rank_offsets = array<std::int64_t>::zeros(rank_count);
+    global_rank_offsets.get_mutable_data()[current_rank] = local_row_count;
+    {
+        ONEDAL_PROFILER_TASK(allreduce_recv_counts, queue);
+        comm.allreduce(global_rank_offsets, spmd::reduce_op::sum).wait();
+    }
+    {
+        ONEDAL_PROFILER_TASK(allreduce_rows_count_global, queue);
+        comm.allreduce(global_row_count, spmd::reduce_op::sum).wait();
+    }
+
+    std::int64_t local_offset = 0;
+
+    for (std::int64_t i = 0; i < current_rank; i++) {
+        ONEDAL_ASSERT(global_rank_offsets.get_data()[i] >= 0);
+        local_offset += global_rank_offsets.get_data()[i];
+    }
+
+    const auto data_nd = pr::table2ndarray<Float>(queue, local_data, sycl::usm::alloc::device);
+    pr::ndarray<Float, 2> data_nd_replace;
+    if (rank_count > 1) {
+        data_nd_replace = pr::ndarray<Float, 2>::empty(queue,
+                                                       { max_local_block_size, column_count },
+                                                       sycl::usm::alloc::device);
+
+        copy(queue, data_nd_replace, data_nd, {}).wait_and_throw();
+    }
+    const auto weights_nd =
+        pr::table2ndarray<Float>(queue, local_weights, sycl::usm::alloc::device);
+    bool use_weights = false;
+    if (weights_nd.get_dimension(0) == data_nd.get_dimension(0)) {
+        use_weights = true;
+    }
+    pr::ndarray<Float, 2> weights_nd_replace;
+    if (rank_count > 1 && use_weights) {
+        weights_nd_replace = pr::ndarray<Float, 2>::empty(queue,
+                                                          { max_local_block_size, 1 },
+                                                          sycl::usm::alloc::device);
+
+        copy(queue, weights_nd_replace, weights_nd, {}).wait_and_throw();
+    }
+    const Float epsilon = desc.get_epsilon() * desc.get_epsilon();
     const std::int64_t min_observations = desc.get_min_observations();
 
+    //array indicates is the point core or no
```

Comment:
Most of the comments in oneDAL begin with a space. Please align the comments in this PR with the rest of the codebase.
 
```suggestion
    // array indicates if the point is a core point or not
```